### DISCUSSION TITLE
How to append/merge files to each other with Gulp

### DISCUSSION THREAD

**Discussion by simonbethke** [gulpjs/gulp#2593] [Help]
I would like to find multiple files with globs and compile one output-file from them. Basically it is:

```
src('/*/file.ini')
  .pipe(createVinylWithModifiedContent())
  .pipe(dest('/output.js', {append:true});
```
Unfortunately, the DestOption "append" seems to be only documented, but not implemented. The gulp-append plugin seems to have a problem if the intermediate vinyls from the inner plugin have not been materialized.

For some context: I am in the process to refactor a grown and old gulpfile.js. In this step, I am also migrating the gulpfile to typescript using also gulpclass. For this task, I am searching for ini-files at different locations for different languages. In the middle of the pipeline, I use the through2 plugin adaptor to parse the ini, modify the keys and create a json-structure from it. In the end I need to write a js-file containing all json-structures combined and wrapped into an "export default ...". As this is part of a refactoring, I don't want to change the general behavior at this stage.

(I have copied this question over here from https://stackoverflow.com/questions/68062945/how-to-append-merge-files-to-each-other-with-gulp and I hope thats okay)

**Comment by yocontra** [comment]
@simonbethke Are you looking for this? https://github.com/gulp-community/gulp-concat

**Reply by simonbethke** [reply]
I have tried that and if i remember correctly, it required the files to be physically existing on the disk. However, I might be mistaking here and will try it again.

**Reply by yocontra** [reply]
@simonbethke Don't think so - it just takes the vinyl files and appends their buffers (or streams) together
